### ISSUE TITLE
Flylint is failing to parse a path correctly

### ISSUE THREAD

**Issue by Wetmelon** [jbenden/vscode-c-cpp-flylint#144]
As of version 1.11.0, Flylint is failing to parse my absolute paths correctly.  It seems to be converting `C:/Tools/API/etc` to `/C:/Tools/API/etc`, and then to `\\C:\\Tools\\API\\etc` (I'm on Windows).  Downgrading to 1.10.2 fixes the problem, so I have a workaround and it's definitely something in the new update.

```
Received a notification to enable and start processing.
validateAllDocuments is starting analysis.
Incoming request for isTrusted property. Have true.
Proxying request for activeConfigName
Sending request to "ms-vstools.cpptools" extension.
Path: ${workspaceFolder}  VALUE: c:\Users\---\Desktop\Test  Globbed is: 
Path: C:/Tools/API/RC10_10_31/V02.11/C-API/Includes  VALUE: C:/Tools/API/RC10_10_31/V02.11/C-API/Includes  Globbed is: C:/Tools/API/RC10_10_31/V02.11/C-API/Includes
Adding system path: /C:/Tools/API/RC10_10_31/V02.11/C-API/Includes
Path: C:/Tools/API/RC10_10_31/V02.11/C-API/Includes/Core  VALUE: C:/Tools/API/RC10_10_31/V02.11/C-API/Includes/Core  Globbed is: C:/Tools/API/RC10_10_31/V02.11/C-API/Includes/Core
Adding system path: /C:/Tools/API/RC10_10_31/V02.11/C-API/Includes/Core
c:\Users\---\Desktop\Test\test.cpp force = false.
c:\Users\---\Desktop\Test\test.cpp is now at version number 1.
Performing lint scan of c:\Users\---\Desktop\Test\test.cpp...
executing:  C:\Program Files\Cppcheck\cppcheck.EXE --inline-suppr --enable=warning,style,performance,portability,information -I /C:/Tools/API/RC10_10_31/V02.11/C-API/Includes -I /C:/Tools/API/RC10_10_31/V02.11/C-API/Includes/Core --std=c11 --std=c++11 -DCONTROLLER_TYPE=RC10_10_31 -DPROGRAM_NAME=RC30_C_Template --language=c++ --platform=native --template="{file}  {line}  {severity} {id}: {message}" c:\Users\---\Desktop\Test\test.cpp
[
  "(information) Couldn't find path given by -I '\\C:\\Tools\\API\\RC10_10_31\\V02.11\\C-API\\Includes\\'",
  "(information) Couldn't find path given by -I '\\C:\\Tools\\API\\RC10_10_31\\V02.11\\C-API\\Includes\\Core\\'",
  'Checking c:\\Users\\---\\Desktop\\Test\\test.cpp ...',
  'Checking c:\\Users\\---\\Desktop\\Test\\test.cpp: CONTROLLER_TYPE=RC10_10_31;PROGRAM_NAME=RC30_C_Template...',
  ''
]
[ '' ]
Completed lint scans...
Skipping analysis because ON_SAVE !== ON_TYPE.
```

**Comment by evanperrygrove-sibelinc** [comment]
I have a similar problem... here is a relevant snippet of my c_cpp_properties:

```json
{
    "configurations": [
        {
            "name": "Currently selected config",
            "includePath": [
                "${workspaceFolder}/foo/bar/interface",
                "${workspaceFolder}/foo/bar/example",
                "${workspaceFolder}/foo/bar/abc",
                "${workspaceFolder}/foo/bar/baz",
                "${default}"
            ]
        }
    ]
}
```

Here is a piece of Flylint 1.11.0's debug output for this set of include paths:

```
Path: ${workspaceFolder}/foo/bar/interface  VALUE: c:\Users\Evan\path\to\project/foo/bar/interface  Globbed is: 
Path: ${workspaceFolder}/foo/bar/example  VALUE: c:\Users\Evan\path\to\project/foo/bar/example  Globbed is: 
Path: ${workspaceFolder}/foo/bar/abc  VALUE: c:\Users\Evan\path\to\project/foo/bar/abc  Globbed is: 
Path: ${workspaceFolder}/foo/bar/baz  VALUE: c:\Users\Evan\path\to\project/foo/bar/baz  Globbed is: 
TypeError: Patterns must be a string (non empty) or an array of strings
    at f (c:\Users\Evan\.vscode\extensions\jbenden.c-cpp-flylint-1.11.0\server\out\server.js:2:55342)
    at AsyncFunction.e.sync (c:\Users\Evan\.vscode\extensions\jbenden.c-cpp-flylint-1.11.0\server\out\server.js:2:55454)
    at c:\Users\Evan\.vscode\extensions\jbenden.c-cpp-flylint-1.11.0\server\out\server.js:2:19926
    at _t (c:\Users\Evan\.vscode\extensions\jbenden.c-cpp-flylint-1.11.0\server\out\server.js:2:103863)
    at Function._s (c:\Users\Evan\.vscode\extensions\jbenden.c-cpp-flylint-1.11.0\server\out\server.js:2:138664)
    at N (c:\Users\Evan\.vscode\extensions\jbenden.c-cpp-flylint-1.11.0\server\out\server.js:2:19855)
```

Two things seem suspicious here.

1. There's nothing following the "Globbed Is:" part of the log messages.
2. The TypeError occurs when the extension would (theoreticaly) reach the `"${default}"` entry.

---

A couple more things that are worthy of note:

A quick search through the repo for the phrase "Globbed Is:" yeilds this log statement:

```js
console.log('Path: ' + ipath + '  VALUE: ' + value + '  Globbed is: ' + globbed_path.toString());
```

So, clearly, there should be *something* following "Globbed Is."

Also, I am very certain that the `"${default}"` entry is the cause of the TypeError. If I move `"${default}"` up in the list of includePaths, the `Path: ... VALUE: ... Globbed Is:` log messages don't print for any of the paths that follow it. Also, if I remove `"${default}"` entirely, the exception doesn't happen at all.

If cpptools is passing the literal string `"${default}"` to Flylint, is that a cpptools bug? It seems to me like cpptools should be responsible for expanding `"${default}"` before passing include paths into another extension, but someone else would know more about this.

**Comment by dzid26** [comment]
@jbenden when is this `/` prefix required?
https://github.com/jbenden/vscode-c-cpp-flylint/blob/f8e7a1b4e34ecfc4826f3c35c77d02a22e79b71c/server/src/server.ts#L471-L474

Both `clang` and `cppcheck` don't like path with the `/` prefix. I am on Windows.
